src/ProcessMaker/Nayra/Bpmn/ParallelGatewayTrait.php

```diff
@@ -0,0 +1,97 @@
+use ProcessMaker\Nayra\Contracts\Bpmn\TokenInterface;
+use ProcessMaker\Nayra\Contracts\Bpmn\TransitionInterface;
+use ProcessMaker\Nayra\Contracts\Repositories\RepositoryFactoryInterface;
+
+/**
+ * Base implementation for a parallel gateway.
+ *
+ * @package ProcessMaker\Nayra\Bpmn
+ */
+trait ParallelGatewayTrait
+{
+
+    use GatewayTrait;
+
+    /**
+     * @var TransitionInterface
+     */
+    private $transition;
+
+    /**
+     * Build the transitions that define the element.
+     *
+     * @param RepositoryFactoryInterface $factory
+     */
+    public function buildTransitions(RepositoryFactoryInterface $factory)
+    {
+        $this->setFactory($factory);
+        $this->transition=new ParallelGatewayTransition($this);
+        $this->transition->attachEvent(TransitionInterface::EVENT_BEFORE_TRANSIT, function()  {
+            $this->notifyEvent(GatewayInterface::EVENT_GATEWAY_ACTIVATED, $this);
+        });
+    }
+
+    /**
+     * Get an input to the element.
+     *
+     * @return StateInterface
+     */
+    public function getInputPlace() {
+        $incomingPlace=new State($this, GatewayInterface::TOKEN_STATE_INCOMMING);
+        $incomingPlace->connectTo($this->transition);
+        $incomingPlace->attachEvent(State::EVENT_TOKEN_ARRIVED, function (TokenInterface $token) {
+            $this->notifyEvent(GatewayInterface::EVENT_GATEWAY_TOKEN_ARRIVES, $this, $token);
+        });
+        $incomingPlace->attachEvent(State::EVENT_TOKEN_CONSUMED, function (TokenInterface $token) {
+            $this->notifyEvent(GatewayInterface::EVENT_GATEWAY_TOKEN_CONSUMED, $this, $token);
+        });
+        return $incomingPlace;
+    }
+
+    /**
+     * Create a connection to a target node.
+     *
+     * @param \ProcessMaker\Nayra\Contracts\Bpmn\FlowNodeInterface $target
+     *
+     * @return $this
+     */
+    protected function buildConnectionTo(FlowNodeInterface $target)
+    {
+        $outgoingPlace = new State($this, GatewayInterface::TOKEN_STATE_OUTGOING);
+        $outgoingTransition = new Transition($this);
+        $outgoingTransition->attachEvent(
+            TransitionInterface::EVENT_AFTER_TRANSIT,
+            function(TransitionInterface $transition)  {
+                $this->notifyEvent(GatewayInterface::EVENT_GATEWAY_TOKEN_PASSED, $this, $transition);
+            }
+        );
+        $this->transition->connectTo($outgoingPlace);
+        $outgoingPlace->connectTo($outgoingTransition);
+        $outgoingTransition->connectTo($target->getInputPlace());
+        return $this;
+    }
+
+    /**
+     * A parallel gateway can not have conditioned outgoing flows.
+     * This method has no effect.
+     *
+     * @param FlowNodeInterface $target
+     * @param callable $condition
+     * @param bool $default
+     *
+     * @return $this
+     *
+     * @codeCoverageIgnore
+     */
+    protected function buildConditionedConnectionTo(FlowNodeInterface $target, callable $condition, $default=false)
```

Comment:
Having dead code is a code smell, avoid forcing  a class to implement methods they don't use

Reply:
Fixed